### PR TITLE
Add middleware

### PR DESCRIPTION
**Pull request by vlas**

### CODE REVIEW

**Comment by marcserrat** [comment]
Please fix code coverage on sonar as exposed via secondary channel.

**Review by jpdolz — APPROVED** [review]
LGTM

**Review by marcserrat — APPROVED** [review]

File: django_telegram/bot/middleware.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class TelegramMiddleware(MiddlewareMixin):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.configs = settings.TELEGRAM_BOT[SETTINGS_MW]
+
+    def get_field_value(self, model, field):
+        if len(field.split('.')) > 1:
```
Comment (ghost) [inline django_telegram/bot/middleware.py]: Better to split just once. Anyway, an iterative implementation can be more clear.

File: django_telegram/bot/middleware.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        view_name = request.resolver_match.view_name
+        config_exists = any(
+            filter(lambda x: x[SETTINGS_MW_VIEW] == view_name, self.configs[SETTINGS_MW_RULES]),
+        )
+        if config_exists:
+            current_config = list(
+                filter(lambda x: x[SETTINGS_MW_VIEW] == view_name, self.configs[SETTINGS_MW_RULES]),
+            )[0]
+            if self.matches_config(current_config, response):
```
Comment (ghost) [inline django_telegram/bot/middleware.py]: This call could raise an exception. Must be handled to ensure the response will always be returned to clients.

**Review by ghost — CHANGES_REQUESTED** [review]

File: django_telegram/bot/middleware.py (hunk 2 above)
Comment (vlas) [reply django_telegram/bot/middleware.py]: fixed

File: django_telegram/bot/middleware.py (hunk 1 above)
Comment (vlas) [reply django_telegram/bot/middleware.py]: changed

**Comment by vlas** [comment]
> Please fix code coverage on sonar as exposed via secondary channel.

fixed thanks

**Review by ghost — APPROVED** [review]

**Pull request merged**